_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -6846,4 +6902,212 @@
     <CActorModel id="AP_ColonyInfestationBuildingDeath" parent="UnitDeathModel">
         <On Terms="ActorCreation" Send="SetTintColor 200,160,133"/>
     </CActorModel>
+    <CActorSound id="AP_SOAHeroicShieldDeathSound" parent="SoundOneShot">
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.Off" Send="Create"/>
+        <Sound value="AP_SOA_GuardianShell_End"/>
+    </CActorSound>
+    <CActorModel id="AP_SOAHeroicShieldLongModel" parent="ModelAnimationStyleContinuous">
+        <Inherits index="BaseModelScale" value="1"/>
+        <Inherits index="Scale" value="1"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsCarrier" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsVoidRay" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsBattlecruiser" Send="Create"/>
+        <On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsHerculesCombine" Send="Create"/>

```

Comment:
I'm surprised there's so many different validators for Heroic Shield against nonprotoss units, I just copied them just in case. 

I'm pretty sure Hercules exists in LOTV, especially on Korhal missions.

Not sure what you want me to do here though. Change "HerculesCombine" to Hercules? And see if Hercules would get guardian shield proc? normally it's not something that'd ever happen.

Reply:
Isn't there a mission where there's allied Jimmy's base?

The problem with the validators is that `IsHerculesCombine` doesn't exist in Liberty.SC2Mod and when you include LotV dependencies for the map, then you have two potentially incompatible implementation of it and then weird stuff happens. You really need to use `AP_IsHercules` that validates against `AP_Hercules` and in ArchipelagoPlayerLotV do the following:

```xml
<On Terms="Behavior.AP_SOAHeroicShieldImmunity.On; ValidateUnit IsHerculesCombine" Send="Create"/>
```
Maybe drop the row with `AP_IsHercules`  
And add `AP_IsHercules` entry to `IsHerculesCombine`

Reply:
Sorry, I'm trying hard to understand it but I don't . I have no idea what you are requesting

Reply:
now it looks OK